### PR TITLE
Infra/ migrate button components to design tokens

### PR DESCRIPTION
**Pull request by lidord-wix**
## Description
Migrate button components to design tokens

## Changelog
Migrate button components to design tokens

### CODE REVIEW

File: src/components/segmentedControl/segment.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ export type SegmentProps = SegmentedControlItemProps & {
  */
 const Segment = React.memo((props: SegmentProps) => {
   const {
-    activeColor = Colors.primary,
+    activeColor = Colors.$textPrimary,
```
Comment (M-i-k-e-l) [inline src/components/segmentedControl/segment.tsx:60]: `activeColor` is used in both text and icon, currently they are the same but I think this should be reflected in the code.
After looking in `SegmentedControl`, why is there a default here as well?

File: src/components/segmentedControl/index.tsx [hunk 2]
```diff
@@ -92,11 +92,11 @@ const SegmentedControl = (props: SegmentedControlProps) => {
     containerStyle,
     style,
     segments,
-    activeColor = Colors.primary,
+    activeColor = Colors.$textPrimary,
```
Comment (M-i-k-e-l) [inline src/components/segmentedControl/index.tsx:95]: `activeColor` is used in both text and icon, currently they are the same but I think this should be reflected in the code.

File: src/components/segmentedControl/index.tsx [hunk 3]
```diff
@@ -201,8 +201,8 @@ const SegmentedControl = (props: SegmentedControlProps) => {
 
 const styles = StyleSheet.create({
   container: {
-    backgroundColor: Colors.grey80,
-    borderColor: Colors.grey60,
+    backgroundColor: Colors.$backgroundNeutralMedium,
```
Comment (M-i-k-e-l) [inline src/components/segmentedControl/index.tsx:204]: I don't see guidelines for this, and `grey80 !== backgroundNeutralMedium`

File: demo/src/screens/componentScreens/ActionBarScreen.tsx [hunk 4]
```diff
@@ -19,12 +19,12 @@ export default class ActionBarScreen extends Component {
 
   render() {
     return (
-      <View flex bg-grey80>
```
Comment (M-i-k-e-l) [inline demo/src/screens/componentScreens/ActionBarScreen.tsx:22]: Something's odd, I don't see the screen as dark, is it ok for you (merge master)?

**Comment by lidord-wix** [comment]
This PR got messy, I'm closing and creating a new one.

**Pull request closed without merging**